plugins/c8y_log_plugin/src/config.rs

```diff
@@ -0,0 +1,730 @@
+use c8y_smartrest::topic::C8yTopic;
+use mqtt_channel::Message;
+use serde::Deserialize;
+use std::borrow::Borrow;
+use std::collections::HashSet;
+use std::fs;
+use std::hash::{Hash, Hasher};
+use std::path::{Path, PathBuf};
+use tracing::{info, warn};
+
+const DEFAULT_PLUGIN_CONFIG_TYPE: &str = "c8y-log-plugin";
+
+#[derive(Deserialize, Debug, Default)]
+#[serde(deny_unknown_fields)]
+struct RawPluginConfig {
+    pub files: Vec<RawFileEntry>,
+}
+
+#[derive(Deserialize, Debug, Default, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct RawFileEntry {
+    pub path: String,
+    #[serde(rename = "type")]
+    config_type: Option<String>,
+}
+
+#[derive(Debug, Eq, PartialEq, Default, Clone)]
+pub struct PluginConfig {
+    pub files: HashSet<FileEntry>,
+}
+
+#[derive(Debug, Eq, Default, Clone)]
+pub struct FileEntry {
+    pub path: String,
+    pub config_type: String,
+}
+
+impl Hash for FileEntry {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.config_type.hash(state);
+    }
+}
+
+impl PartialEq for FileEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.config_type == other.config_type
```

Comment:
This will cause two entries with the same config type to replace each other. Didn't you want to allow multiple entries with the same `type`?

Reply:
as above